lib/weather_web/controllers/data_controller.ex

```diff
@@ -1,20 +1,18 @@
 defmodule WeatherWeb.DataController do
   use WeatherWeb, :controller
 
-  alias Weather.Stations
   alias Weather.Stations.Data
 
   action_fallback(WeatherWeb.FallbackController)
 
   def create(conn, %{"data" => data_params, "station_id" => station_id}) do
-    params =
-      data_params
-      |> Map.put("station_id", station_id)
+    params = for {key, val} <- data_params, into: %{}, do: {String.to_atom(key), val}
+    data = %Data{}
+    data = %{data | fields: Map.merge(data.fields, params)}
+    data = %{data | tags: %{data.tags | station_id: String.to_integer(station_id)}}
```

Comment:
There probably is a more functional way to sanitize the data received from a station for Influx!?